In [19]:
import pandas as pd

In [20]:
wnv = pd.read_csv('wnv_aggr_final.csv') #The WNV data given by Dr.Aiman
se = pd.read_csv('se_data.csv') #The Socio-Economic data that was cleaned by Vinu
rc = pd.read_csv('race_data.csv')
rohan = pd.read_csv("./Poverty/ACS_12_5YR_S1701/ACS_12_5YR_S1701_with_ann.csv",encoding='ISO-8859-1',low_memory=False)

In [21]:
#Getting the idea of the wnv dataset
print(wnv.columns)
print("\nShape: ", wnv.shape)
print("\nYears: ", wnv.YEAR.unique())
print("\nCounties: ", len(wnv.GEOID10.unique()))
wnv.head(5)

Index(['YEAR', 'GEOID10', 'CTYNAME', 'STUSPS', 'DeG12Region', 'Total_BG.sBG',
       'County_WNV_Class', 'County_WNV_Class2', 'POPESTIMATE', 'NI_', 'NNI_',
       'NI_IR_', 'Establish_', 'NIIR_Z', 'TEMPERATURE', 'HUMIDITY',
       'PRECIPITATION', 'CLIMATE_ZONE'],
      dtype='object')

Shape:  (14484, 18)

Years:  [1999 2000 2001 2002 2003 2004 2005 2006 2007 2008 2009 2010 2011 2012
 2013 2014 2015]

Counties:  852


YEAR  GEOID10         CTYNAME STUSPS  DeG12Region  Total_BG.sBG  \
0  1999     1001  Autauga County     AL            2            15   
1  1999     1003  Baldwin County     AL            2            28   
2  1999     1015  Calhoun County     AL            2            45   
3  1999     1033  Colbert County     AL            2            24   
4  1999     1045     Dale County     AL            2            10   

   County_WNV_Class County_WNV_Class2 POPESTIMATE  NI_  NNI_  NI_IR_  \
0                 2              NI-S       42963    0     0    0.00   
1                 4        NI-M-pre12      137555    0     0    0.00   
2                 2          NI-S-NNI      114910    0     0    0.00   
3                 1              none       54715    0     0    0.00   
4                 2          NI-S-NNI       49118    0     0    0.00   

   Establish_ NIIR_Z  TEMPERATURE  HUMIDITY  PRECIPITATION CLIMATE_ZONE  
0           0      0       291.52     73.04           3.56           se  
1           0      0       294.36     75.95           3.23           se  
2           0      0       290.08     72.00           3.03           se  
3           0      0       290.19     72.07           2.87           se  
4           0      0       292.77     72.05           3.14           se

In [22]:
print(se.columns)
print("\nShape: ", se.shape)
print("\nYears: ", se.YEAR.unique())
print("\nCounties: ", len(se.STCOU.unique()))
se.head(5)

Index(['YEAR', 'Areaname', 'STCOU', 'Median_Household_Income',
       'Poverty_number_of_people', 'Poverty_percent_of_people',
       'Public_school_enrollment_Fall'],
      dtype='object')

Shape:  (44772, 7)

Years:  [1995 1997 1998 1999 2000 2001 2002 2003 2004 2005 2006 2007 2008 2009]

Counties:  3198


YEAR       Areaname  STCOU  Median_Household_Income  \
0  1995  UNITED STATES      0                    34076   
1  1995        ALABAMA   1000                    27357   
2  1995    Autauga, AL   1001                    34842   
3  1995    Baldwin, AL   1003                    35014   
4  1995    Barbour, AL   1005                    24423   

   Poverty_number_of_people  Poverty_percent_of_people  \
0                  36424609                      13.80   
1                    752097                      17.60   
2                      5471                      13.40   
3                     15494                      12.40   
4                      6423                      24.90   

   Public_school_enrollment_Fall  
0                       43369917  
1                         730509  
2                           7381  
3                          19384  
4                           5111

In [23]:
se['Calculated_population'] = (se['Poverty_number_of_people'] * 100)/se['Poverty_percent_of_people']
pd.options.display.float_format = '{:.2f}'.format
se.head(5)

YEAR       Areaname  STCOU  Median_Household_Income  \
0  1995  UNITED STATES      0                    34076   
1  1995        ALABAMA   1000                    27357   
2  1995    Autauga, AL   1001                    34842   
3  1995    Baldwin, AL   1003                    35014   
4  1995    Barbour, AL   1005                    24423   

   Poverty_number_of_people  Poverty_percent_of_people  \
0                  36424609                      13.80   
1                    752097                      17.60   
2                      5471                      13.40   
3                     15494                      12.40   
4                      6423                      24.90   

   Public_school_enrollment_Fall  Calculated_population  
0                       43369917           263946442.03  
1                         730509             4273278.41  
2                           7381               40828.36  
3                          19384              124951.61  
4                           5111               25795.18

In [24]:
print(rc.columns)
print("\nShape: ", rc.shape)
print("\nYears: ", rc.YEAR.unique())
print("\nCounties: ", len(rc.STCOU.unique()))
rc.head(5)

Index(['YEAR', 'Areaname', 'STCOU', 'Resident_population_White_alone',
       'Resident_population_White_alone_percent',
       'Resident_population_Black_alone',
       'Resident_population_Black_alone_percent',
       'Resident_population_American_Indian_and_Alaska_Native_alone',
       'Resident_population_American_Indian_and_Alaska_Native_alone_percent',
       'Resident_population_Asian_alone',
       'Resident_population_Asian_alone_percent',
       'Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone',
       'Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone_percent',
       'Resident_population_Two_or_more_races',
       'Resident_population_Two_or_more_races_percent',
       'Resident_population_Hispanic_or_Latino_Origin',
       'Resident_population_Hispanic_or_Latino_Origin_percent',
       'Resident_population_Not_Hispanic_White_alone',
       'Resident_population_Not_Hispanic_White_alone_percent'],
      dtype='object')

Shape:  (31980,

YEAR       Areaname  STCOU  Resident_population_White_alone  \
0  2000  UNITED STATES      0                        228602436   
1  2000        ALABAMA   1000                          3197029   
2  2000    Autauga, AL   1001                            35549   
3  2000    Baldwin, AL   1003                           124101   
4  2000    Barbour, AL   1005                            15151   

   Resident_population_White_alone_percent  Resident_population_Black_alone  \
0                                    81.00                         35807824   
1                                    71.80                          1161485   
2                                    81.00                             7558   
3                                    87.80                            14678   
4                                    52.20                            13482   

   Resident_population_Black_alone_percent  \
0                                    12.70   
1                                    26.10   
2                                    17.20   
3                                    10.40   
4                                    46.40   

   Resident_population_American_Indian_and_Alaska_Native_alone  \
0                                            2675900             
1                                              23038             
2                                                198             
3                                                823             
4                                                131             

   Resident_population_American_Indian_and_Alaska_Native_alone_percent  \
0                                               0.90                     
1                                               0.50                     
2                                               0.50                     
3                                               0.60                     
4                                               0.50                     

   Resident_population_Asian_alone  Resident_population_Asian_alone_percent  \
0                         10687312                                     3.80   
1                            32446                                     0.70   
2                              204                                     0.50   
3                              547                                     0.40   
4                               87                                     0.30   

   Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone  \
0                                             465614                      
1                                               1531                      
2                                                 13                      
3                                                 38                      
4                                                  9                      

   Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone_percent  \
0                                               0.20                              
1                                               0.00                              
2                                               0.00                              
3                                               0.00                              
4                                               0.00                              

   Resident_population_Two_or_more_races  \
0                                3932871   
1                                  36320   
2                                    350   
3                                   1171   
4                                    175   

   Resident_population_Two_or_more_races_percent  \
0                                           1.40   
1                                           0.80   
2                                           0.80   
3                                           0.80   
4                                           0.60   

   Reside

In [25]:
rc['Total_number'] = rc['Resident_population_White_alone'] + rc['Resident_population_Black_alone'] + rc['Resident_population_American_Indian_and_Alaska_Native_alone'] + rc['Resident_population_Asian_alone'] + rc['Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone'] + rc['Resident_population_Two_or_more_races'] 
rc['Total_percent'] = rc['Resident_population_White_alone_percent'] + rc['Resident_population_Black_alone_percent'] + rc['Resident_population_American_Indian_and_Alaska_Native_alone_percent'] + rc['Resident_population_Asian_alone_percent'] + rc['Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone_percent'] + rc['Resident_population_Two_or_more_races_percent'] 
rc.head(5)


YEAR       Areaname  STCOU  Resident_population_White_alone  \
0  2000  UNITED STATES      0                        228602436   
1  2000        ALABAMA   1000                          3197029   
2  2000    Autauga, AL   1001                            35549   
3  2000    Baldwin, AL   1003                           124101   
4  2000    Barbour, AL   1005                            15151   

   Resident_population_White_alone_percent  Resident_population_Black_alone  \
0                                    81.00                         35807824   
1                                    71.80                          1161485   
2                                    81.00                             7558   
3                                    87.80                            14678   
4                                    52.20                            13482   

   Resident_population_Black_alone_percent  \
0                                    12.70   
1                                    26.10   
2                                    17.20   
3                                    10.40   
4                                    46.40   

   Resident_population_American_Indian_and_Alaska_Native_alone  \
0                                            2675900             
1                                              23038             
2                                                198             
3                                                823             
4                                                131             

   Resident_population_American_Indian_and_Alaska_Native_alone_percent  \
0                                               0.90                     
1                                               0.50                     
2                                               0.50                     
3                                               0.60                     
4                                               0.50                     

   Resident_population_Asian_alone  ...  \
0                         10687312  ...   
1                            32446  ...   
2                              204  ...   
3                              547  ...   
4                               87  ...   

   Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone  \
0                                             465614                      
1                                               1531                      
2                                                 13                      
3                                                 38                      
4                                                  9                      

   Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone_percent  \
0                                               0.20                              
1                                               0.00                              
2                                               0.00                              
3                                               0.00                              
4                                               0.00                              

   Resident_population_Two_or_more_races  \
0                                3932871   
1                                  36320   
2                                    350   
3                                   1171   
4                                    175   

   Resident_population_Two_or_more_races_percent  \
0                                           1.40   
1                                           0.80   
2                                           0.80   
3                                           0.80   
4                                           0.60   

   Resident_population_Hispanic_or_Latino_Origin  \
0                                       35642379   
1                                          76504   
2                                            610   
3                

In [26]:
for i in rohan.iloc[0]:
    print(i)

Id
Id2
Geography
Total; Estimate; Population for whom poverty status is determined
Total; Margin of Error; Population for whom poverty status is determined
Below poverty level; Estimate; Population for whom poverty status is determined
Below poverty level; Margin of Error; Population for whom poverty status is determined
Percent below poverty level; Estimate; Population for whom poverty status is determined
Percent below poverty level; Margin of Error; Population for whom poverty status is determined
Total; Estimate; AGE - Under 18 years
Total; Margin of Error; AGE - Under 18 years
Below poverty level; Estimate; AGE - Under 18 years
Below poverty level; Margin of Error; AGE - Under 18 years
Percent below poverty level; Estimate; AGE - Under 18 years
Percent below poverty level; Margin of Error; AGE - Under 18 years
Total; Estimate; AGE - Under 18 years - Related children under 18 years
Total; Margin of Error; AGE - Under 18 years - Related children under 18 years
Below poverty level; E

In [27]:
wnv_2000 = wnv[wnv['YEAR'] == 2000]
se_2000 = se[se['YEAR'] == 2000]
rc_2000 = rc[rc['YEAR'] == 2000]

In [28]:
print(wnv_2000.shape)
wnv_2000[['YEAR' , 'GEOID10', 'CTYNAME', 'POPESTIMATE']].head(5)

(852, 18)


YEAR  GEOID10         CTYNAME POPESTIMATE
852  2000     1001  Autauga County       44021
853  2000     1003  Baldwin County      141342
854  2000     1015  Calhoun County      111081
855  2000     1033  Colbert County       54997
856  2000     1045     Dale County       49124

In [29]:
se_2000[['YEAR','STCOU','Areaname','Calculated_population']].head(6)

YEAR  STCOU       Areaname  Calculated_population
12792  2000      0  UNITED STATES           279478637.17
12793  2000   1000        ALABAMA             4363828.77
12794  2000   1001    Autauga, AL               44580.95
12795  2000   1003    Baldwin, AL              142835.05
12796  2000   1005    Barbour, AL               26337.84
12797  2000   1007       Bibb, AL               20005.62

In [30]:
print(rc_2000.shape)
rc_2000[['YEAR','STCOU','Areaname','Total_number']].head(6)

(3198, 21)


YEAR  STCOU       Areaname  Total_number
0  2000      0  UNITED STATES     282171957
1  2000   1000        ALABAMA       4451849
2  2000   1001    Autauga, AL         43872
3  2000   1003    Baldwin, AL        141358
4  2000   1005    Barbour, AL         29035
5  2000   1007       Bibb, AL         19936

In [31]:
result = pd.merge(wnv_2000, rc_2000, how='inner', left_on='GEOID10', right_on='STCOU')

In [32]:
print(result.shape)
result.columns

(852, 39)


Index(['YEAR_x', 'GEOID10', 'CTYNAME', 'STUSPS', 'DeG12Region', 'Total_BG.sBG',
       'County_WNV_Class', 'County_WNV_Class2', 'POPESTIMATE', 'NI_', 'NNI_',
       'NI_IR_', 'Establish_', 'NIIR_Z', 'TEMPERATURE', 'HUMIDITY',
       'PRECIPITATION', 'CLIMATE_ZONE', 'YEAR_y', 'Areaname', 'STCOU',
       'Resident_population_White_alone',
       'Resident_population_White_alone_percent',
       'Resident_population_Black_alone',
       'Resident_population_Black_alone_percent',
       'Resident_population_American_Indian_and_Alaska_Native_alone',
       'Resident_population_American_Indian_and_Alaska_Native_alone_percent',
       'Resident_population_Asian_alone',
       'Resident_population_Asian_alone_percent',
       'Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone',
       'Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone_percent',
       'Resident_population_Two_or_more_races',
       'Resident_population_Two_or_more_races_percent',
       'R

In [33]:
result = result[['YEAR_x', 'GEOID10', 'CTYNAME', 'POPESTIMATE','Total_number']]
result["POPESTIMATE"] = pd.to_numeric(result["POPESTIMATE"])
result.head(5)

YEAR_x  GEOID10         CTYNAME  POPESTIMATE  Total_number
0    2000     1001  Autauga County        44021         43872
1    2000     1003  Baldwin County       141342        141358
2    2000     1015  Calhoun County       111081        111342
3    2000     1033  Colbert County        54997         55024
4    2000     1045     Dale County        49124         49068

In [34]:
result.dtypes

YEAR_x           int64
GEOID10          int64
CTYNAME         object
POPESTIMATE      int64
Total_number     int64
dtype: object

In [35]:
result['Error'] = (result["POPESTIMATE"] - result["Total_number"]).abs()
result['Percentage_error'] = (result["Error"]/result["POPESTIMATE"])*100

print("Values with greater than 10% error: \n\n",result[result.Percentage_error > 10.0 ].count())

#Sorting by percentage error
result.sort_values(by=['Percentage_error'], ascending=False)

Values with greater than 10% error: 

 YEAR_x              0
GEOID10             0
CTYNAME             0
POPESTIMATE         0
Total_number        0
Error               0
Percentage_error    0
dtype: int64


YEAR_x  GEOID10               CTYNAME  POPESTIMATE  Total_number  Error  \
745    2000    51019        Bedford County        66897         60606   6291   
280    2000    21163          Meade County        28189         26523   1666   
771    2000    51540  Charlottesville city        41351         40117   1234   
389    2000    28073          Lamar County        40211         39335    876   
177    2000    13219         Oconee County        25874         26368    494   
..      ...      ...                   ...          ...           ...    ...   
111    2000    12031          Duval County       779800        779803      3   
74     2000     6107         Tulare County       368627        368628      1   
70     2000     6095         Solano County       396974        396974      0   
313    2000    24001       Allegany County        74804         74804      0   
285    2000    22005      Ascension Parish        77335         77335      0   

     Percentage_error  
745              9.40  
280              5.91  
771              2.98  
389              2.18  
177              1.91  
..                ...  
111              0.00  
74               0.00  
70               0.00  
313              0.00  
285              0.00  

[852 rows x 7 columns]

In [36]:
print("Values with greater than 5% error: \n\n",result[result.Percentage_error > 2.0 ].count())

#Sorting by percentage error
result.sort_values(by=['Percentage_error'], ascending=False)

Values with greater than 5% error: 

 YEAR_x              4
GEOID10             4
CTYNAME             4
POPESTIMATE         4
Total_number        4
Error               4
Percentage_error    4
dtype: int64


YEAR_x  GEOID10               CTYNAME  POPESTIMATE  Total_number  Error  \
745    2000    51019        Bedford County        66897         60606   6291   
280    2000    21163          Meade County        28189         26523   1666   
771    2000    51540  Charlottesville city        41351         40117   1234   
389    2000    28073          Lamar County        40211         39335    876   
177    2000    13219         Oconee County        25874         26368    494   
..      ...      ...                   ...          ...           ...    ...   
111    2000    12031          Duval County       779800        779803      3   
74     2000     6107         Tulare County       368627        368628      1   
70     2000     6095         Solano County       396974        396974      0   
313    2000    24001       Allegany County        74804         74804      0   
285    2000    22005      Ascension Parish        77335         77335      0   

     Percentage_error  
745              9.40  
280              5.91  
771              2.98  
389              2.18  
177              1.91  
..                ...  
111              0.00  
74               0.00  
70               0.00  
313              0.00  
285              0.00  

[852 rows x 7 columns]

#### For year 2009:

In [47]:
wnv_2009 = wnv[wnv['YEAR'] == 2009]
se_2009 = se[se['YEAR'] == 2009]
rc_2009 = rc[rc['YEAR'] == 2009]

In [48]:
print(wnv_2009.shape)
wnv_2009[['YEAR' , 'GEOID10', 'CTYNAME', 'POPESTIMATE']].head(5)

(852, 18)


YEAR  GEOID10         CTYNAME POPESTIMATE
8520  2009     1001  Autauga County       54135
8521  2009     1003  Baldwin County      179406
8522  2009     1015  Calhoun County      118363
8523  2009     1033  Colbert County       54426
8524  2009     1045     Dale County       49854

#### Verifying manually for Champaign county to determine which population is the closest

#### For year 2000:

In [43]:
wnv_2000[wnv_2000['GEOID10'] == 17019]

YEAR  GEOID10           CTYNAME STUSPS  DeG12Region  Total_BG.sBG  \
1045  2000    17019  Champaign County     IL            1           106   

      County_WNV_Class County_WNV_Class2 POPESTIMATE  NI_  NNI_  NI_IR_  \
1045                 4        NI-M-pre12      180041    0     0    0.00   

      Establish_ NIIR_Z  TEMPERATURE  HUMIDITY  PRECIPITATION CLIMATE_ZONE  
1045           0              285.45     74.88           2.84            c

In [46]:
se_2000[se_2000['STCOU'] == 17019]

YEAR       Areaname  STCOU  Median_Household_Income  \
13411  2000  Champaign, IL  17019                    39603   

       Poverty_number_of_people  Poverty_percent_of_people  \
13411                     19506                      11.70   

       Public_school_enrollment_Fall  Calculated_population  
13411                          24193              166717.95

In [45]:
rc_2000[rc_2000['STCOU'] == 17019]

YEAR       Areaname  STCOU  Resident_population_White_alone  \
619  2000  Champaign, IL  17019                           144228   

     Resident_population_White_alone_percent  Resident_population_Black_alone  \
619                                    80.20                            20304   

     Resident_population_Black_alone_percent  \
619                                    11.30   

     Resident_population_American_Indian_and_Alaska_Native_alone  \
619                                                497             

     Resident_population_American_Indian_and_Alaska_Native_alone_percent  \
619                                               0.30                     

     Resident_population_Asian_alone  ...  \
619                            12068  ...   

     Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone  \
619                                                 92                      

     Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone_percent  \
619                                               0.10                              

     Resident_population_Two_or_more_races  \
619                                   2742   

     Resident_population_Two_or_more_races_percent  \
619                                           1.50   

     Resident_population_Hispanic_or_Latino_Origin  \
619                                           5288   

     Resident_population_Hispanic_or_Latino_Origin_percent  \
619                                               2.90       

     Resident_population_Not_Hispanic_White_alone  \
619                                        139567   

     Resident_population_Not_Hispanic_White_alone_percent  Total_number  \
619                                              77.60           179931   

     Total_percent  
619         100.10  

[1 rows x 21 columns]

> Actual population:                 179936

> Population from WNV file:    180041 (Difference = 105,   + 0.05%)

> Population from SE file:       166718 (Difference = 13218, - 7.34%)

>Population from Race file:   179931 (Difference = 5,     - 0.002%)

#### For year 2009:

In [49]:
wnv_2009[wnv_2009['GEOID10'] == 17019]

YEAR  GEOID10           CTYNAME STUSPS  DeG12Region  Total_BG.sBG  \
8713  2009    17019  Champaign County     IL            1           106   

      County_WNV_Class County_WNV_Class2 POPESTIMATE  NI_  NNI_  NI_IR_  \
8713                 4        NI-M-pre12      199968    0     0    0.00   

      Establish_            NIIR_Z  TEMPERATURE  HUMIDITY  PRECIPITATION  \
8713           1  -.14449740034662       285.10     76.69           3.26   

     CLIMATE_ZONE  
8713            c

In [50]:
se_2009[se_2009['STCOU'] == 17019]

YEAR       Areaname  STCOU  Median_Household_Income  \
42193  2009  Champaign, IL  17019                    42101   

       Poverty_number_of_people  Poverty_percent_of_people  \
42193                     34982                      19.90   

       Public_school_enrollment_Fall  Calculated_population  
42193                          23479              175788.94

In [52]:
rc_2009[rc_2009['STCOU'] == 17019]

YEAR       Areaname  STCOU  Resident_population_White_alone  \
29401  2009  Champaign, IL  17019                           152686   

       Resident_population_White_alone_percent  \
29401                                    78.00   

       Resident_population_Black_alone  \
29401                            23450   

       Resident_population_Black_alone_percent  \
29401                                    12.00   

       Resident_population_American_Indian_and_Alaska_Native_alone  \
29401                                                611             

       Resident_population_American_Indian_and_Alaska_Native_alone_percent  \
29401                                               0.30                     

       Resident_population_Asian_alone  ...  \
29401                            15040  ...   

       Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone  \
29401                                                103                      

       Resident_population_Native_Hawaiian_and_Other_Pacific_Islander_alone_percent  \
29401                                               0.10                              

       Resident_population_Two_or_more_races  \
29401                                   3781   

       Resident_population_Two_or_more_races_percent  \
29401                                           1.90   

       Resident_population_Hispanic_or_Latino_Origin  \
29401                                           8438   

       Resident_population_Hispanic_or_Latino_Origin_percent  \
29401                                               4.30       

       Resident_population_Not_Hispanic_White_alone  \
29401                                        145137   

       Resident_population_Not_Hispanic_White_alone_percent  Total_number  \
29401                                              74.20           195671   

       Total_percent  
29401         100.00  

[1 rows x 21 columns]

> Actual population:                 196323

> Population from WNV file:    199968 (Difference = 3645,   + 1.85%)

> Population from SE file:       175789 (Difference = 20534, - 10.45%)

> Population from Race file:   195671 (Difference = 652,     - 0.33%)

#### Based on this analysis, we decided that the population data in the Race file is the closest to the actual values.

#### We will use the population data for the years 1995 to 1999 from Socio-economic data file.

#### We will use the population data for the years 2000 to 2009 from Race data file.

#### We will use the population data for the years 2010 to 2015 from WNV data file.
